### PR TITLE
Refactor more classes to use Page helper methods

### PR DESCRIPTION
**Pull request by pettyjamesm**
Comparable to changes in https://github.com/prestosql/presto/pull/5218

Two commits:
- Adds `Page#extractChannel(int)` helper method to allow efficient single channel Page extraction
- Refactors more classes and operators to use application `Page` class helper methods which avoid extra allocations and copies of Block arrays

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-common/src/main/java/com/facebook/presto/common/Page.java [hunk 1]
```diff
@@ -329,6 +329,11 @@ public Page copyPositions(int[] retainedPositions, int offset, int length)
         return wrapBlocksWithoutCopy(length, blocks);
     }
 
+    public Page extractChannel(int channel)
+    {
+        return wrapBlocksWithoutCopy(positionCount, new Block[]{this.blocks[channel]});
```
Comment (highker) [inline presto-common/src/main/java/com/facebook/presto/common/Page.java:334]: Will this save CPU c.p. `extractChannels(new int[]{channel});`?

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java [hunk 2]
```diff
@@ -550,14 +550,8 @@ public Type getFinalType()
         @Override
         public void addInput(GroupByIdBlock groupIdsBlock, Page page)
         {
-            Block[] blocks = new Block[page.getChannelCount() + 1];
-            for (int i = 0; i < page.getChannelCount(); i++) {
-                blocks[i] = page.getBlock(i);
-            }
-            // Add group id block
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java:557]: keep this comment to the second to last line

**Review by highker — APPROVED** [review]

File: presto-common/src/main/java/com/facebook/presto/common/Page.java (hunk 1 above)
Comment (pettyjamesm) [reply presto-common/src/main/java/com/facebook/presto/common/Page.java:334]: They should be the same IFF the method is fully inlined and the temporary int array becomes stack allocated, but that seemed more fragile to depend on compared to just providing the desired implementation directly.

File: presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java (hunk 2 above)
Comment (pettyjamesm) [reply presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java:557]: Woops, good catch. Added it back.

**Pull request merged**